src/Adapter/MSTest.TestAdapter/Discovery/TypeEnumerator.cs

```diff
@@ -119,12 +117,11 @@ internal Collection<UnitTestElement> GetTests(ICollection<string> warnings)
             currentType = currentType.BaseType;
         }
 
-        return new Collection<UnitTestElement>(
-            tests.GroupBy(
-                t => t.TestMethod.Name,
-                (_, elements) =>
-                    elements.OrderBy(t => inheritanceDepths[t.TestMethod.DeclaringClassFullName ?? t.TestMethod.FullClassName]).First())
-                .ToList());
+        return tests.GroupBy(
```

Comment:
This was created a `List<UnitTestElement>`, then wrapped it in a `Collection<UnitTestElement>`. Now that we return `List<UnitTestElement>`, there is no need to create an extra object.